src/crypto/CHIPCryptoPALOpenSSL.cpp

```diff
@@ -1970,6 +1970,37 @@ CHIP_ERROR ExtractAKIDFromX509Cert(const ByteSpan & certificate, MutableByteSpan
     return ExtractKIDFromX509Cert(false, certificate, akid);
 }
 
+CHIP_ERROR ExtractSerialNumberFromX509Cert(const ByteSpan & certificate, MutableByteSpan & serialNumber)
+{
+    CHIP_ERROR err                        = CHIP_NO_ERROR;
+    X509 * x509certificate                = nullptr;
+    const unsigned char * pCertificate    = certificate.data();
```

Comment:
This is assuming that `unsigned char` and `uint8_t` is the same type.  Is that actually guaranteed?

Might be better as:
```suggestion
    auto * pCertificate    = Uint8::to_const_uchar(certificate.data());
```